### PR TITLE
Add missing scp_component options to idrac_server_config_profile

### LINKED ISSUE #342: Manage EventFilters
##### Description
I would like to use ansible to manage the EventFilters

##### New or existing component
I think it's a new component, it is exported in SCPs.

`"FQDD": "EventFilters.*"`

Direct use case for me, I want to configure memory mail warnings. so `MEM_1_2#Alert#Email` I want to set to enabled.

#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by Klaas-**
~~depends on dell/omsdk#23~~  see https://github.com/dell/dellemc-openmanage-ansible-modules/pull/343#issuecomment-1772181059

##### SUMMARY
add missing scp_component options to idrac_server_config_profile
First step for #342 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
idrac_server_config_profile

##### ISSUES
References but does not fix #342

##### OUTPUT

```paste below
no test cases changed, currently only testcases for idrac scp_component exist
```

### CODE REVIEW

**Comment by Klaas-** [comment]
Forgot to add signoff as requested in dco, I would consider this a very minor update though :)

**Comment by Klaas-** [comment]
During my testing (scp_component eventfilters) I noticed that scp does not properly apply when using:
`shutdown_type: NoReboot`
it leads to the apply job being paused with 20% with message "SYS296: No reboot Server Configuration Profile Import job scheduled, Waiting for System Reboot to complete the operation."

When using:
`shutdown_type: Graceful`
the scp is job is being executed properly, the system is not rebooted. I am guessing this is not an ansible issue but how the idrac works. Should this be reported to Dell or is this viewed as a documentation only issue?

**Review by anupamaloke — COMMENTED** [review]
LGTM

**Comment by Klaas-** [comment]
@anupamaloke who would I contact about the problem with `shutdown_type: NoReboot` from comment https://github.com/dell/dellemc-openmanage-ansible-modules/pull/343#issuecomment-981111762 ? Should that be handled in a support case?

**Comment by Klaas-** [comment]
hmm okay, I think I'll PR something for the docs of the ansible module then. But that also means there is no "do not reboot" option? and I need to check somehow if graceful will reboot the server or not

File: plugins/modules/idrac_server_config_profile.py [hunk 1]
```diff
@@ -64,8 +64,11 @@
       - If C(BIOS), this module exports or imports BIOS configuration from SCP file.
       - If C(NIC), this module exports or imports NIC configuration from SCP file.
       - If C(RAID), this module exports or imports RAID configuration from SCP file.
+      - If C(EVENTFILTERS), this module exports or imports EVENTFILTERS configuration from SCP file.
+      - If C(SUPPORTASSIST), this module exports or imports SUPPORTASSIST configuration from SCP file.
+      - If C(LIFECYCLECONTROLLER), this module exports or imports LIFECYCLECONTROLLER configuration from SCP file.
     type: str
-    choices: ['ALL', 'IDRAC', 'BIOS', 'NIC', 'RAID']
+    choices: ['ALL', 'IDRAC', 'BIOS', 'NIC', 'RAID', 'EVENTFILTERS', 'SUPPORTASSIST', 'LIFECYCLECONTROLLER']
```
Comment (sachin-apa) [inline plugins/modules/idrac_server_config_profile.py:72]: @Klaas- we checked with iDRAC team on the options, `SUPPORTASSIST` is not part of the supported component options.
Can you please point us to an reference where you have seen the support for the option  `SUPPORTASSIST`?
@anupamaloke

Comment (Klaas-) [reply plugins/modules/idrac_server_config_profile.py:72]: you can see those in a complete export of "ALL" :)

Comment (Klaas-) [reply plugins/modules/idrac_server_config_profile.py:72]: so an "ALL" export shows something like this (I grepped for FQDD in the file):
```
                "FQDD": "SupportAssist.Embedded.1",
                "FQDD": "RAID.Integrated.1-1",
                        "FQDD": "Disk.Virtual.0:RAID.Integrated.1-1",
                        "FQDD": "Enclosure.Internal.0-1:RAID.Integrated.1-1",
                                "FQDD": "Disk.Bay.0:Enclosure.Internal.0-1:RAID.Integrated.1-1",
                                "FQDD": "Disk.Bay.1:Enclosure.Internal.0-1:RAID.Integrated.1-1",
                                "FQDD": "Disk.Bay.2:Enclosure.Internal.0-1:RAID.Integrated.1-1",
                "FQDD": "NIC.Slot.3-1-1",
                "FQDD": "NIC.Slot.3-2-1",
                "FQDD": "NIC.Slot.4-1-1",
                "FQDD": "NIC.Slot.4-2-1",
                "FQDD": "NIC.Integrated.1-1-1",
                "FQDD": "NIC.Integrated.1-2-1",
                "FQDD": "NIC.Integrated.1-3-1",
                "FQDD": "NIC.Integrated.1-4-1",
                "FQDD": "FC.Slot.2-2",
                "FQDD": "FC.Slot.2-1",
                "FQDD": "FC.Slot.1-2",
                "FQDD": "FC.Slot.1-1",
                "FQDD": "BIOS.Setup.1-1",
                "FQDD": "iDRAC.Embedded.1",
                "FQDD": "System.Embedded.1",
                "FQDD": "LifecycleController.Embedded.1",
                "FQDD": "EventFilters.SystemHealth.1",
                "FQDD": "EventFilters.Storage.1",
                "FQDD": "EventFilters.Updates.1",
                "FQDD": "EventFilters.Audit.1",
                "FQDD": "EventFilters.Configuration.1",
                "FQDD": "EventFilters.WorkNotes.1",
```

**Comment by sameer1046** [comment]
Can one of the admins verify this patch?

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

**Review by sachin-apa — DISMISSED** [review]

**Review by sachin-apa — APPROVED** [review]

**Comment by Klaas-** [comment]
Hi @sachin-apa 
I don't think this is mergeable without the PR in https://github.com/dell/omsdk/pull/23

**Comment by Klaas-** [comment]
At least that is what I wrote ~2 years ago in https://github.com/dell/dellemc-openmanage-ansible-modules/pull/343#issue-1065034585

**Comment by sachin-apa** [comment]
> Hi @sachin-apa I don't think this is mergeable without the PR in [dell/omsdk#23](https://github.com/dell/omsdk/pull/23)

@Klaas-  this module has evolved over 2 years !!   it is now independent of omsdk, we have few more enhancement planned in this month, we will merge this PR and add those enhancements/fixes on top of it..

**Comment by Klaas-** [comment]
> > Hi @sachin-apa I don't think this is mergeable without the PR in [dell/omsdk#23](https://github.com/dell/omsdk/pull/23)
> 
> @Klaas- this module has evolved over 2 years !! it is now independent of omsdk, we have few more enhancement planned in this month, we will merge this PR and add those enhancements/fixes on top of it..

That's good to hear, I haven't followed it closely I have to admit :) I guess I'll just strike the reference from the initial comment then

**Review by Jennifer-John — APPROVED** [review]

**Pull request merged**